### DISCUSSION TITLE
Making changes to the in-game UI in Lazer

### DISCUSSION THREAD

**Discussion by Sprfrost** [ppy/osu#17043] [Ideas]
It would be cool if we could hide the progress bar (or customize it). Make the fps counter customizable and the key overlay too (the one like in the new cpol videos that would be sick).

**Comment by TheFantasticLoki** [comment]
The overlay he is referencing is actually very clean and since he did not provide a specific example of said overlay here is a link to it:  https://www.youtube.com/watch?v=ALlQINurqFU

The software being used to achieve this is called Gosumemory and is a very popular software that also allows you to use these overlays inside the game itself on stable: https://github.com/l3lackShark/gosumemory

**Reply by TheFantasticLoki** [reply]
The overlays themselves are based on HTML, JS and CSS code while pulling values from gosumemory which grabs the memory values from stable.

**Reply by peppy** [reply]
I'm pretty sure you linked the wrong video. Also a screenshot would be provided so we don't have to access youtube to view proposals.

**Reply by TheFantasticLoki** [reply]
I believe giving players a built in alternative would be doable and probably more performant since gosumemory's gameoverlay does add a fair amount of input delay depending on the size of the overlay. The reasoning behind this is due to the fact it is constantly rendering transparent pixels alongside opaque pixels and I feel you could implement a better alternative while supporting the use of these web overlays outside the game for streaming in case the player finds them too distracting in game.

But overall having the ability to use html css and js to customize the game is more of an enthusiast/tinker oriented player feature but could include an official listing of overlays players can download and use giving a less sketchy way of using these tools.

**Comment by peppy** [comment]
The end goal is to have every element customisable to any degree. That's what the skinning system was designed for.

**Reply by TheFantasticLoki** [reply]
Here is a screenshot showing an exapmle of the key overlay he was mentioning: 
![image](https://user-images.githubusercontent.com/26889568/156303294-5956c4ea-6db1-4975-b917-e506055ea186.png)
